navapbc/wic-participant-recertification-portal
# PRP-227 - Magical routing!

## Ticket

https://wicmtdp.atlassian.net/browse/PRP-227

## Changes
* URLs are validated to check that there is a valid Local Agency in them
* Excess / incorrect paths are redirected to real paths
* Invalid sessions redirect to the same clinic (if valid)
* Navigating to / bumps the individual to `/gallatin/recertify` (the first agency in the database)
* Added tests for redirect logic
* Added playwright tests for redirect logic, cookie validation
* Updated seed script to support atomic runs (rerunning it won't hurt)
* Removed hardcoded clinic string from cookie parser
* Centralized all logic for routing in utils/redirect.ts

## Testing
1.  run `npm run dev`
2. navigate to `http://localhost:3000/bogus/url/somewhere`
3. See `http://localhost:3000/gallatin/recertify` in your browser

More exciting / advanced tests are available. Real routes should still work for valid clinics, for example
(http://localhost:3000/gallatin/recertify/intro) or (http://localhost:3000/missoula/recertify/intro)
